filesystem/File.php

```diff
@@ -238,7 +238,7 @@ public function exists() {
 	 */
 	public static function find($filename) {
 		// Get the base file if $filename points to a resampled file
-		$filename = preg_replace('/_resampled\/[^-]+-/', '', $filename);
+		$filename = preg_replace('/_resampled\/(.+\/|[^-]+-)/', '', $filename);
```

Comment:
It is backwards compatible though: old filenames (both before and after the base64 change) will match the second part of that "or" relation. New filenames will match the first part - which may in fact have slashes in them because the image may have been transformed multiple times (e.g. a rotate and then a translate), resulting in `_resampled/[rotate]/[translate]/orig_filename.jpg`. Since we want everything before `orig_filename.jpg` to be gone, it does need to match that 'middle' slash as well :)

As for putting it in a new method, sure, I could do that. A static method in the Image class perhaps?